python/lib/modeldata/dmod/modeldata/hydrofabric/geopackage_hydrofabric.py

```diff
@@ -0,0 +1,593 @@
+import fiona
+import geopandas as gpd
+import hashlib
+from pandas.util import hash_pandas_object
+from pathlib import Path
+from typing import Callable, Dict, FrozenSet, Iterable, List, Optional, Tuple, Union
+from hypy import Catchment, Nexus, Realization
+from .hydrofabric import Hydrofabric
+from ..subset import SubsetDefinition
+
+
+class GeoPackageCatchment(Catchment):
+    """
+    Customized subtype of ::class:`Catchment` backed by dataframes from a parent ::class:`GeoPackageHydrofabric`.
+
+    Type overrides most properties of superclass so that those can be evaluated on-demand. This supports creating an
+    instance based on data and a parent, and defers the association (or really, associates indirectly and gets the
+    linked object on-demand) to things like a connected nexus for which the nexus object may or may not yet exist.
+    """
+
+    __slots__ = ["_cat_id", "_hydrofabric", "_catchments_df", "_nexuses_df", "_realization", "_col_cat_id",
+                 "_col_nex_id", "_col_to_cat", "_col_to_nex"]
+
+    def __init__(self, cat_id: str, hydrofabric: 'GeoPackageHydrofabric', catchments_df: gpd.GeoDataFrame,
+                 nexuses_df: gpd.GeoDataFrame, col_cat_id: str, col_nex_id: str, col_to_cat: str, col_to_nex: str):
+        """
+        Initialize this instance.
+
+        Parameters
+        ----------
+        cat_id : str
+            The id of the represented catchment.
+        hydrofabric : GeoPackageHydrofabric
+            The backing package hydrofabric.
+        catchments_df : gpd.GeoDataFrame
```

Comment:
The `catchments_df` is from the `divides` layer, yes.  

I didn't go out of my way to go back and include a link to the documentation because the documentation was inconsistent with what I saw in the data.  I don't remember exactly what I ran into off the top of my head; the differences were subtle but enough that I got hung up on them a few times.  I had to load a few geopackages and dataframes manually to figure out what was going on.

I'll probably add a link and a warning of potential discrepancies in the next PR;  there will be a good bit more work related to this stuff in the near term.

Reply:
Yeah, ive noticed inconsistencies as well. Sounds good to me.